interop-testing/src/test/java/io/grpc/testing/integration/CompressionTest.java

```diff
@@ -0,0 +1,1466 @@
+ * is willing to decode, a second RPC is sent to show that the client has learned and will use
+ * the encoding.  This applies to all tests that follow the "..E .A." pattern.
+ *
+ * <p> In cases where compression is negotiated, but either the client or the server doesn't
+ * actually want to encode, a dummy codec is used to record usage.  If compression is not enabled,
+ * the codec will see no data pass through.  This is checked on each test to ensure the code is
+ * doing the right thing.
+ *
+ * <p>Lastly, tests are organized according to the table above, in little-endian naming order.  More
+ * featureful clients and servers are represented near the bottom, while less featureful ones are
+ * near the top.
+ */
+@RunWith(JUnit4.class)
+public class CompressionTest {
+  private static final ScheduledExecutorService executor = Executors.newScheduledThreadPool(2);
+  // Ensures that boththe request and response messages are more than 0 bytes.  The framer/deframer
+  // may not use the compressor if the message is empty.
+  private static final SimpleRequest REQUEST = SimpleRequest.newBuilder()
+      .setResponseSize(1)
+      .build();
+
+  private Fzip clientCodec = new Fzip();
+  private Fzip serverCodec = new Fzip();
+  private DecompressorRegistry clientDecompressors = DecompressorRegistry.newEmptyInstance();
+  private DecompressorRegistry serverDecompressors = DecompressorRegistry.newEmptyInstance();
+  private CompressorRegistry clientCompressors = CompressorRegistry.newEmptyInstance();
+  private CompressorRegistry serverCompressors = CompressorRegistry.newEmptyInstance();
+
+  /** The headers received by the server from the client */
+  private volatile Metadata serverResponseHeaders;
+  /** The headers received by the client from the server */
+  private volatile Metadata clientResponseHeaders;
+  private volatile boolean enableClientMessageCompression;
+  private volatile boolean enableServerMessageCompression;
+
+  private Server server;
+  private TestServiceBlockingStub stub;
+
+  @Before
+  public void setUp() throws Exception {
+    int serverPort = TestUtils.pickUnusedPort();
+    server = ServerBuilder.forPort(serverPort)
+        .addService(ServerInterceptors.intercept(
+            TestServiceGrpc.bindService(new TestServiceImpl(executor)),
+            new ServerCompressorInterceptor()))
+        .compressorRegistry(serverCompressors)
+        .decompressorRegistry(serverDecompressors)
+        .build()
+        .start();
+
+    ManagedChannel channel = ManagedChannelBuilder.forAddress("localhost", serverPort)
+        .decompressorRegistry(clientDecompressors)
+        .compressorRegistry(clientCompressors)
+        .intercept(new ClientCompresorInterceptor())
+        .usePlaintext(true)
+        .build();
+    stub = TestServiceGrpc.newBlockingStub(channel);
+  }
+
+  @After
+  public void tearDown() {
+    server.shutdownNow();
+    executor.shutdownNow();
+  }
+
+  @Test
+  public void compression_cae_cae() {
```

Comment:
Because verification is harder than just setting the booleans. Also, this approach has a the benefit that you can read the test method and know what it does.  There is no need to look into another method (which would likely be complicated) to see what it does.

This approach is really easy to read.